src/Network/AWS/ARN.hs

```diff
@@ -158,33 +161,29 @@ _ARN = prism' fromARN toARN
 
 -- | Split a 'Text' into colon-separated parts.
 --
--- This is not truly a lawful 'Iso'', but it is useful. The 'Iso''
--- laws are violated for lists whose members contain @':'@:
---
--- >>> [":"] ^. from colons . colons
--- ["",""]
---
--- The laws are also violated on empty lists:
---
--- >>> [] ^. from colons . colons
--- [""]
---
--- However, it is still a useful tool:
+-- This is useful for editing the resource part of an ARN:
 --
 -- >>> "foo:bar:baz" & colons . ix 1 .~ "quux"
 -- "foo:quux:baz"
-colons :: Iso' Text [Text]
-colons = iso (T.splitOn ":") (T.intercalate ":")
+--
+-- Writing back through the lens ignores the string it is applied to:
+--
+-- >>> "Hello, world!" & colons .~ "dude" :| ["sweet"]
+-- "dude:sweet"
+colons :: Lens' Text (NonEmpty Text)
+colons f t =
+  T.intercalate ":" . NonEmpty.toList
+    <$> f (NonEmpty.fromList (T.splitOn ":" t))
```

Comment:
Why not use `lens`?

Reply:
I have this vague hope that I can one day not commit to any particular optic library because of `generic-lens`/`generic-optics`. That would require writing out the `_ARN` prism by hand and mean depending on `profunctors` instead of `lens`.

Reply:
I like that idea but have some thoughts:
- I think you should either do the whole change or none of it.
- It could be worthwhile to define `lens`, `Lens'`, `prism'`, and `Prism'` locally.
- I really wish there were [a version of microlens with Prisms](https://github.com/stevenfontanella/microlens/issues/105).

Reply:
@Tristano8 and I smashed this out this morning.